Pokedex async redux

giantiyan opened the pull request.

giantiyan committed:
Activity #1.2

giantiyan committed:
Details Screen

raelsj committed:
lint checker

raelsj committed:
all branch PR

raelsj committed:
removed working directory

giantiyan committed:
Details 2

giantiyan committed:
Pokemon details #3

giantiyan committed:
Merge branch 'pokemon-details' into test

giantiyan committed:
pokedex 1

giantiyan committed:
fixed some lints

raelsj commented on .github/workflows/lint_workflow.yml:
```diff
@@ -0,0 +1,40 @@
+          channel: stable
+          architecture: x64
+      
+      - name: Install packages and analyze
```
divide this to 2 steps this should only run `flutter pub get` so you should have 3 steps

- pub get
- build runner
- analyze

raelsj commented on lib/api/handlers/pokemon_handler.dart:
```diff
@@ -0,0 +1,25 @@
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+
+import '../../utilities/constants.dart';
```
complete the imports

raelsj commented on lib/api/handlers/pokemon_handler.dart:
```diff
@@ -0,0 +1,25 @@
+
+class PokemonHandler {
+  static Future<List<PokemonModel>?> getPokemon() async {
+    var response = http.Response('', 100);
```
this is fine for now, but in the long run we will be using Dio

raelsj commented on lib/api/models/pokemon_model.dart:
```diff
@@ -0,0 +1,14 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
```
nice 💪🏻

raelsj commented on lib/api/models/pokemon_model.freezed.dart:
```diff
@@ -0,0 +1,194 @@
+// coverage:ignore-file
```
this .freezed.dart and .g.dart files should not be committed, add these to your `.gitignore` files

raelsj commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,145 @@
+            SizedBox(
+                height: 250,
+                child: Image.network(
+                    'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${pokemon?.url.toString().split('/')[6]}.png')),
```
this should be an extension or a method inside this class

raelsj commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,145 @@
+                        Padding(
+                          padding: const EdgeInsets.all(12.0),
+                          child: Text(
+                            toBeginningOfSentenceCase(pokemon?.name).toString(),
```
ask @SaturnoMark or @ryle-roasa how to make an extension for this

raelsj commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,145 @@
+                          child: Text(
+                            toBeginningOfSentenceCase(pokemon?.name).toString(),
+                            style: const TextStyle(
+                                fontSize: 22, fontWeight: FontWeight.bold),
```
trailing comma

raelsj commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,145 @@
+                                labelPadding: EdgeInsets.symmetric(horizontal: 10.0),
+                                tabs: <Widget>[
+                                  Tab(
+                                    text: 'About',
```
create a constant file for these strings

raelsj commented on lib/feature/home_page/home_page.dart:
```diff
@@ -0,0 +1,152 @@
+
+  final List<PokemonModel>? pokemon;
+
+  // @override
```
remove

raelsj commented on lib/feature/details_page/details_page_connector.dart:
```diff
@@ -0,0 +1,24 @@
+// import 'package:async_redux/async_redux.dart';
```
you should not commit these kind of files unless you have TODOs all over it

raelsj commented on lib/feature/details_page/details_page_vm.dart:
```diff
@@ -0,0 +1,24 @@
+// import 'package:async_redux/async_redux.dart';
```
same above

raelsj commented on lib/feature/home_page/home_page.dart:
```diff
@@ -0,0 +1,152 @@
+            mainAxisSpacing: 10),
+        itemCount: pokemon?.length,
+        itemBuilder: (context, index) {
+          var id = pokemon?[index].url.toString().split('/')[6];
```
I've seen this code already, meaning it is DRY(don't repeat yourself) violation. you probably really need an extension for this

raelsj commented on lib/feature/home_page/home_page.dart:
```diff
@@ -0,0 +1,152 @@
+
+          return InkWell(
+            onTap: () {
+              Navigator.push(
```
@SaturnoMark @ryle-roasa teach him proper routing

raelsj commented on lib/feature/home_page/home_page.dart:
```diff
@@ -0,0 +1,152 @@
+                  MaterialPageRoute(
+                      builder: (context) => DetailsPage(
+                            pokemon: pokemon?[index],
+                          )));
```
🤔 proper format

raelsj commented on lib/state/app_state.freezed.dart:
```diff
@@ -0,0 +1,160 @@
+// coverage:ignore-file
```
same comment regarding these files

raelsj commented on pubspec.yaml:
```diff
@@ -0,0 +1,95 @@
+name: pokedex
```
kindly remove all comments in this file

raelsj commented on test/widget_test.dart:
```diff
@@ -0,0 +1,29 @@
+// This is a basic Flutter widget test.
```
you can remove this file, unless you want to add a unit test for this app

raelsj commented:
@giantiyan the lints.

giantiyan committed:
pr feedbacks

giantiyan committed:
types and colors

ryle-roasa commented on lib/feature/home_page/widgets/type.dart:
```diff
@@ -0,0 +1,31 @@
+            horizontal: 12,
+            vertical: 5,
+          ),
+          margin: EdgeInsets.only(bottom: 8),
```
Add **right** property since it will also be used horizontally in the details page

giantiyan committed:
[UI] Show Pokemon About - Show Pokemon Base Stats

ryle-roasa commented on lib/feature/home_page/home_page.dart:
```diff
@@ -0,0 +1,145 @@
+                        Column(
+                          children: [
+                            ...?pokemon?[index].types?.map((type) => Type(type.name)),
+                            // for (var i = 0; i < int.parse(pokemon?[index].types?.length.toString() ?? ''); i++) ...[
```
remove comments

ryle-roasa commented on lib/feature/home_page/home_page_connector.dart:
```diff
@@ -0,0 +1,23 @@
+        onInit: (store) async => store.dispatch(GetPokemonAction()),
+        builder: (context, vm) {
+          return HomePage(
+            pokemon: vm.pokemon,
```
remove trailing comma

ryle-roasa commented on lib/feature/home_page/home_page_vm.dart:
```diff
@@ -0,0 +1,22 @@
+  @override
+  Vm fromStore() {
+    return HomePageVm(
+      pokemon: state.pokemon,
```
remove trailing comma

ryle-roasa commented on lib/feature/home_page/home_page_vm.dart:
```diff
@@ -0,0 +1,22 @@
+
+class HomePageVm extends Vm {
+  HomePageVm({
+    this.pokemon,
```
remove trailing comma

ryle-roasa commented on lib/feature/home_page/widgets/type.dart:
```diff
@@ -0,0 +1,31 @@
+            horizontal: 12,
+            vertical: 5,
+          ),
+          margin: EdgeInsets.only(bottom: 8, right: 8),
```
add trailing comma

ryle-roasa commented on lib/feature/moves_tab/moves_tab_connector.dart:
```diff
@@ -0,0 +1,21 @@
+class MovesTabConnector extends StatelessWidget {
+  const MovesTabConnector(this.pokemonID);
+
+  final int? pokemonID;
```
rename to pokemonId

ryle-roasa commented on lib/feature/moves_tab/moves_tab_vm.dart:
```diff
@@ -0,0 +1,17 @@
+class MovesTabVmFactory extends VmFactory<AppState, MovesTabConnector> {
+  @override
+  Vm fromStore() => MovesTabVm(
+    moves: state.moves
```
try auto formatting if this goes up to line 8

ryle-roasa commented on lib/pokedex_app.dart:
```diff
@@ -0,0 +1,22 @@
+      store: store,
+      child: MaterialApp(
+        title: 'Flutter Demo',
+        theme: ThemeData(primarySwatch: Colors.blue),
```
you can remove line 16 and 17

ryle-roasa commented on lib/state/actions/details_page_actions.dart:
```diff
@@ -0,0 +1,43 @@
+import 'package:async_redux/async_redux.dart';
+import 'package:pokedex/api/handlers/pokemon_handler.dart';
+import 'package:pokedex/state/app_state.dart';
+
```
Could you add a documentation for each action like:

```
/// Get pokemon about details and save it to the state
class GetPokemonAboutAction extends ReduxAction<AppState> {
```

ryle-roasa replied:
Then rename pokemonID -> pokemonId

ryle-roasa commented on lib/state/app_state.dart:
```diff
@@ -0,0 +1,24 @@
+import 'package:pokedex/api/models/pokemon_moves_model.dart';
+
+part 'app_state.freezed.dart';
+
```
remove space

ryle-roasa commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,137 @@
+
+class DetailsPage extends StatelessWidget {
+  const DetailsPage({
+    required this.pokemon,
```
remove trailing comma

ryle-roasa commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,137 @@
+    return Scaffold(
+      backgroundColor: pokemon?.types?[0].name.toString().pokemonColor,
+      appBar: AppBar(
+        backgroundColor: Colors.red,
```
remove trailing comma

ryle-roasa commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,137 @@
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: pokemon?.types?[0].name.toString().pokemonColor,
```
use .first instead of [0]

`.types?.first`

ryle-roasa commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,137 @@
+                    ),
+                    Row(
+                      children: [
+                        ...?pokemon?.types?.map((type) => Type(type.name)),
```
remove trailing comma

ryle-roasa commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,137 @@
+                  color: Colors.white,
+                  borderRadius: BorderRadius.only(
+                    topLeft: Radius.circular(25),
+                    topRight: Radius.circular(25),
```
replace with `borderRadius: BorderRadius.vertical(top: Radius.circular(25))`

ryle-roasa commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,137 @@
+                    )
+                  ],
+                ),
+                child: Column(
```
Column not needed

ryle-roasa commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,137 @@
+                      child: Column(
+                        mainAxisSize: MainAxisSize.min,
+                        children: [
+                          TabBar(indicatorColor: Colors.orangeAccent,
```
format, indicatorColor move to next line

ryle-roasa commented on lib/feature/details_page/details_page.dart:
```diff
@@ -0,0 +1,137 @@
+                              Tab(text: moves),
+                            ],
+                          ),
+                          Container(
```
Replace Container widget with Expanded

giantiyan committed:
[UI & Enhancement]

- Show Pokemon Moves
- Create Pokemon Tile Widget
- Get Pokemon Types in PokemonTile
- Show Pokemon Types in Details Screen
- Show Filter Modal Bottom
